backend/Models/auth/updateUser.js

```diff
@@ -0,0 +1,96 @@
+const {pool} = require('../db');
+const {email} = require('../../utils');
+const otplib = require('otplib');
+
+/**
+ *
+ * @param {*} param0
+ * @param {String} param0.username
+ * @param {String} param0.email
+ * @param {String} param0.name
+ * @param {Number} param0.branch
+ * @param {Number} param0.department
+ * @param {String} param0.admission_no
+ * @param {Number} param0.semester
+ * @return {Promise}
+ *
+ */
+function updateUser({
+  username,
+  email: emailId,
+  name,
+  branch,
+  department,
+  admission_no: admissionNo,
+  semester,
+}) {
+  return new Promise((resolve, reject) => {
+    const secret = otplib.authenticator.generateSecret();
+    const otp = otplib.authenticator.generate(secret);
+    let query = `UPDATE users SET `;
+    let arr = [];
+    if (name) {
+      query += `name=?,`;
+      arr.push(name);
+    }
+    if (branch) {
+      if (department || admissionNo || semester || emailId) {
```

Comment:
This code looks dirty, instead check if previously anyone has sent any parameter and then append ```,``` before the current parameter.

Reply:
I have made changes..